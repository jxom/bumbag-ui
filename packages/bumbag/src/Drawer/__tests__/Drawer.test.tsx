import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Drawer } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Drawer.State baseId="test" visible modal={false}>
        {(drawer) => (
          <div>
            <Drawer.Disclosure {...drawer}>Toggle</Drawer.Disclosure>
            <Drawer {...drawer}>This is a side overlay</Drawer>
          </div>
        )}
      </Drawer.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for an invisible overlay', () => {
    const { container } = render(
      <Drawer.State baseId="test" modal={false}>
        {(drawer) => (
          <div>
            <Drawer.Disclosure {...drawer}>Toggle</Drawer.Disclosure>
            <Drawer {...drawer}>This is a side overlay</Drawer>
          </div>
        )}
      </Drawer.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <Drawer.State baseId="test" modal={false}>
        {(drawer) => (
          <div>
            <Drawer.Disclosure {...drawer}>Toggle</Drawer.Disclosure>
            <Drawer {...drawer} backgroundColor="primary">
              This is a side overlay
            </Drawer>
          </div>
        )}
      </Drawer.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly without backdrop', () => {
    const { container } = render(
      <Drawer.State baseId="test" modal={false}>
        {(drawer) => (
          <div>
            <Drawer.Disclosure {...drawer}>Toggle</Drawer.Disclosure>
            <Drawer {...drawer} hideBackdrop>
              This is a side overlay
            </Drawer>
          </div>
        )}
      </Drawer.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for full screen', () => {
    const { container } = render(
      <Drawer.State baseId="test" visible modal={false}>
        {(drawer) => (
          <div>
            <Drawer.Disclosure {...drawer}>Toggle</Drawer.Disclosure>
            <Drawer {...drawer} isFullScreen>
              This is a side overlay
            </Drawer>
          </div>
        )}
      </Drawer.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('Drawer.styles.base should render correctly', () => {
    const { container } = render(
      <Drawer.State baseId="test" visible modal={false}>
        {(drawer) => (
          <div>
            <Drawer.Disclosure {...drawer}>Toggle</Drawer.Disclosure>
            <Drawer {...drawer} overrides={{ Drawer: { styles: { base: { backgroundColor: 'red' } } } }}>
              This is a side overlay
            </Drawer>
          </div>
        )}
      </Drawer.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Drawer.Disclosure.styles.base should render correctly', () => {
    const { container } = render(
      <Drawer.State baseId="test" visible modal={false}>
        {(drawer) => (
          <div>
            <Drawer.Disclosure
              {...drawer}
              overrides={{ Drawer: { Disclosure: { styles: { base: { backgroundColor: 'red' } } } } }}
            >
              Toggle
            </Drawer.Disclosure>
            <Drawer {...drawer}>This is a side overlay</Drawer>
          </div>
        )}
      </Drawer.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Drawer.styles.base should render correctly', () => {
    const { container } = render(
      <Drawer.State baseId="test" visible modal={false}>
        {(drawer) => (
          <div>
            <Drawer.Disclosure {...drawer}>Toggle</Drawer.Disclosure>
            <Drawer {...drawer}>This is a side overlay</Drawer>
          </div>
        )}
      </Drawer.State>,
      { theme: { Drawer: { styles: { base: { backgroundColor: 'red' } } } } }
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Drawer.Disclosure.styles.base should render correctly', () => {
    const { container } = render(
      <Drawer.State baseId="test" visible modal={false}>
        {(drawer) => (
          <div>
            <Drawer.Disclosure {...drawer}>Toggle</Drawer.Disclosure>
            <Drawer {...drawer}>This is a side overlay</Drawer>
          </div>
        )}
      </Drawer.State>,
      { theme: { Drawer: { Disclosure: { styles: { base: { backgroundColor: 'red' } } } } } }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(
      <Drawer.State baseId="test" visible modal={false}>
        {(drawer) => (
          <div>
            <Drawer.Disclosure {...drawer}>Toggle</Drawer.Disclosure>
            <Drawer {...drawer}>This is a side overlay</Drawer>
          </div>
        )}
      </Drawer.State>,
      { theme: { Drawer: { defaultProps: { className: 'test' } } } }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
