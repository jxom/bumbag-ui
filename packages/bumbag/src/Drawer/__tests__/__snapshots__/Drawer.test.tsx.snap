// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-7 {
  position: fixed;
  z-index: 19900410;
  min-width: 320px;
  top: 50%;
  left: 1.25em;
  -webkit-transform: translate3d(0px,-50%,0px);
  -ms-transform: translate3d(0px,-50%,0px);
  transform: translate3d(0px,-50%,0px);
}

@media screen and (max-width:320px) {
  .emotion-7 {
    min-width: unset;
    width: 100%;
  }
}

.emotion-7:focus {
  outline: none;
}

.emotion-6 {
  box-shadow: 0px 3px 12px 0px rgba(0,0,0,0.15),0px 0px 6px 3px rgba(0,0,0,0.02);
}

.emotion-6.emotion-6 {
  position: fixed;
  background-color: white;
  left: 0;
  height: 100vh;
}

.emotion-3 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ModalDisclosure fp-DrawerDisclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-6 emotion-7 emotion-1 emotion-2 test fp-Box fp-Drawer fp-Modal"
    data-dialog="true"
    id="test"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="emotion-3 emotion-1 emotion-2 test fp-Box fp-ModalBackdrop"
      data-dialog-ref="test"
    >
      <div />
    </div>
    This is a side overlay
  </div>
</div>
`;

exports[`overrides Drawer.Disclosure.styles.base should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-8 {
  position: fixed;
  z-index: 19900410;
  min-width: 320px;
  top: 50%;
  left: 1.25em;
  -webkit-transform: translate3d(0px,-50%,0px);
  -ms-transform: translate3d(0px,-50%,0px);
  transform: translate3d(0px,-50%,0px);
}

@media screen and (max-width:320px) {
  .emotion-8 {
    min-width: unset;
    width: 100%;
  }
}

.emotion-8:focus {
  outline: none;
}

.emotion-7 {
  box-shadow: 0px 3px 12px 0px rgba(0,0,0,0.15),0px 0px 6px 3px rgba(0,0,0,0.02);
}

.emotion-7.emotion-7 {
  position: fixed;
  background-color: white;
  left: 0;
  height: 100vh;
}

.emotion-4 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-ModalDisclosure fp-DrawerDisclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-7 emotion-8 emotion-2 emotion-3 fp-Box fp-Drawer fp-Modal"
    data-dialog="true"
    id="test"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="emotion-4 emotion-2 emotion-3 fp-Box fp-ModalBackdrop"
      data-dialog-ref="test"
    >
      <div />
    </div>
    This is a side overlay
  </div>
</div>
`;

exports[`overrides Drawer.styles.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

.emotion-7 {
  position: fixed;
  z-index: 19900410;
  min-width: 320px;
  top: 50%;
  left: 1.25em;
  -webkit-transform: translate3d(0px,-50%,0px);
  -ms-transform: translate3d(0px,-50%,0px);
  transform: translate3d(0px,-50%,0px);
}

@media screen and (max-width:320px) {
  .emotion-7 {
    min-width: unset;
    width: 100%;
  }
}

.emotion-7:focus {
  outline: none;
}

.emotion-7.emotion-7 {
  background-color: #ff0000;
}

.emotion-6 {
  box-shadow: 0px 3px 12px 0px rgba(0,0,0,0.15),0px 0px 6px 3px rgba(0,0,0,0.02);
}

.emotion-6.emotion-6 {
  position: fixed;
  background-color: white;
  left: 0;
  height: 100vh;
}

.emotion-6.emotion-6 {
  background-color: #ff0000;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ModalDisclosure fp-DrawerDisclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-6 emotion-7 emotion-1 emotion-2 fp-Box fp-Drawer fp-Modal"
    data-dialog="true"
    id="test"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="emotion-3 emotion-1 emotion-2 fp-Box fp-ModalBackdrop"
      data-dialog-ref="test"
    >
      <div />
    </div>
    This is a side overlay
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-7 {
  position: fixed;
  z-index: 19900410;
  min-width: 320px;
  top: 50%;
  left: 1.25em;
  -webkit-transform: translate3d(0px,-50%,0px);
  -ms-transform: translate3d(0px,-50%,0px);
  transform: translate3d(0px,-50%,0px);
}

@media screen and (max-width:320px) {
  .emotion-7 {
    min-width: unset;
    width: 100%;
  }
}

.emotion-7:focus {
  outline: none;
}

.emotion-6 {
  box-shadow: 0px 3px 12px 0px rgba(0,0,0,0.15),0px 0px 6px 3px rgba(0,0,0,0.02);
}

.emotion-6.emotion-6 {
  position: fixed;
  background-color: white;
  left: 0;
  height: 100vh;
}

.emotion-3 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ModalDisclosure fp-DrawerDisclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-6 emotion-7 emotion-1 emotion-2 fp-Box fp-Drawer fp-Modal"
    data-dialog="true"
    id="test"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="emotion-3 emotion-1 emotion-2 fp-Box fp-ModalBackdrop"
      data-dialog-ref="test"
    >
      <div />
    </div>
    This is a side overlay
  </div>
</div>
`;

exports[`props should render correctly for an invisible overlay 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-7 {
  position: fixed;
  z-index: 19900410;
  min-width: 320px;
  top: 50%;
  left: 1.25em;
  -webkit-transform: translate3d(0px,-50%,0px);
  -ms-transform: translate3d(0px,-50%,0px);
  transform: translate3d(0px,-50%,0px);
}

@media screen and (max-width:320px) {
  .emotion-7 {
    min-width: unset;
    width: 100%;
  }
}

.emotion-7:focus {
  outline: none;
}

.emotion-6 {
  box-shadow: 0px 3px 12px 0px rgba(0,0,0,0.15),0px 0px 6px 3px rgba(0,0,0,0.02);
}

.emotion-6.emotion-6 {
  position: fixed;
  background-color: white;
  left: 0;
  height: 100vh;
}

.emotion-3 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ModalDisclosure fp-DrawerDisclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-6 emotion-7 emotion-1 emotion-2 fp-Box fp-Drawer fp-Modal"
    data-dialog="true"
    hidden=""
    id="test"
    role="dialog"
    style="display: none;"
    tabindex="-1"
  >
    <div
      class="emotion-3 emotion-1 emotion-2 fp-Box fp-ModalBackdrop"
      data-dialog-ref="test"
      hidden=""
      style="display: none;"
    >
      <div />
    </div>
    This is a side overlay
  </div>
</div>
`;

exports[`props should render correctly for full screen 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-7 {
  position: fixed;
  z-index: 19900410;
  min-width: 320px;
  top: 50%;
  left: 1.25em;
  -webkit-transform: translate3d(0px,-50%,0px);
  -ms-transform: translate3d(0px,-50%,0px);
  transform: translate3d(0px,-50%,0px);
}

@media screen and (max-width:320px) {
  .emotion-7 {
    min-width: unset;
    width: 100%;
  }
}

.emotion-7:focus {
  outline: none;
}

.emotion-3 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

.emotion-6 {
  box-shadow: 0px 3px 12px 0px rgba(0,0,0,0.15),0px 0px 6px 3px rgba(0,0,0,0.02);
}

.emotion-6.emotion-6 {
  position: fixed;
  background-color: white;
  left: 0;
  height: 100vh;
  width: 100%;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ModalDisclosure fp-DrawerDisclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-6 emotion-7 emotion-1 emotion-2 fp-Box fp-Drawer fp-Modal"
    data-dialog="true"
    id="test"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="emotion-3 emotion-1 emotion-2 fp-Box fp-ModalBackdrop"
      data-dialog-ref="test"
    >
      <div />
    </div>
    This is a side overlay
  </div>
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-7 {
  position: fixed;
  z-index: 19900410;
  min-width: 320px;
  top: 50%;
  left: 1.25em;
  -webkit-transform: translate3d(0px,-50%,0px);
  -ms-transform: translate3d(0px,-50%,0px);
  transform: translate3d(0px,-50%,0px);
}

@media screen and (max-width:320px) {
  .emotion-7 {
    min-width: unset;
    width: 100%;
  }
}

.emotion-7:focus {
  outline: none;
}

.emotion-6 {
  box-shadow: 0px 3px 12px 0px rgba(0,0,0,0.15),0px 0px 6px 3px rgba(0,0,0,0.02);
}

.emotion-6.emotion-6 {
  position: fixed;
  background-color: white;
  left: 0;
  height: 100vh;
}

.emotion-3 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

.emotion-9 {
  background-color: #574feb !important;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ModalDisclosure fp-DrawerDisclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-6 emotion-7 emotion-1 emotion-9 fp-Box fp-Drawer fp-Modal"
    data-dialog="true"
    hidden=""
    id="test"
    role="dialog"
    style="display: none;"
    tabindex="-1"
  >
    <div
      class="emotion-3 emotion-1 emotion-2 fp-Box fp-ModalBackdrop"
      data-dialog-ref="test"
      hidden=""
      style="display: none;"
    >
      <div />
    </div>
    This is a side overlay
  </div>
</div>
`;

exports[`props should render correctly without backdrop 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  position: fixed;
  z-index: 19900410;
  min-width: 320px;
  top: 50%;
  left: 1.25em;
  -webkit-transform: translate3d(0px,-50%,0px);
  -ms-transform: translate3d(0px,-50%,0px);
  transform: translate3d(0px,-50%,0px);
}

@media screen and (max-width:320px) {
  .emotion-4 {
    min-width: unset;
    width: 100%;
  }
}

.emotion-4:focus {
  outline: none;
}

.emotion-3 {
  box-shadow: 0px 3px 12px 0px rgba(0,0,0,0.15),0px 0px 6px 3px rgba(0,0,0,0.02);
}

.emotion-3.emotion-3 {
  position: fixed;
  background-color: white;
  left: 0;
  height: 100vh;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ModalDisclosure fp-DrawerDisclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-3 emotion-4 emotion-1 emotion-2 fp-Box fp-Drawer fp-Modal"
    data-dialog="true"
    hidden=""
    id="test"
    role="dialog"
    style="display: none;"
    tabindex="-1"
  >
    This is a side overlay
  </div>
</div>
`;

exports[`theming Drawer.Disclosure.styles.base should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-8 {
  position: fixed;
  z-index: 19900410;
  min-width: 320px;
  top: 50%;
  left: 1.25em;
  -webkit-transform: translate3d(0px,-50%,0px);
  -ms-transform: translate3d(0px,-50%,0px);
  transform: translate3d(0px,-50%,0px);
}

@media screen and (max-width:320px) {
  .emotion-8 {
    min-width: unset;
    width: 100%;
  }
}

.emotion-8:focus {
  outline: none;
}

.emotion-7 {
  box-shadow: 0px 3px 12px 0px rgba(0,0,0,0.15),0px 0px 6px 3px rgba(0,0,0,0.02);
}

.emotion-7.emotion-7 {
  position: fixed;
  background-color: white;
  left: 0;
  height: 100vh;
}

.emotion-4 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

.emotion-0.emotion-0 {
  background-color: #ff0000;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-ModalDisclosure fp-DrawerDisclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-7 emotion-8 emotion-2 emotion-3 fp-Box fp-Drawer fp-Modal"
    data-dialog="true"
    id="test"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="emotion-4 emotion-2 emotion-3 fp-Box fp-ModalBackdrop"
      data-dialog-ref="test"
    >
      <div />
    </div>
    This is a side overlay
  </div>
</div>
`;

exports[`theming Drawer.styles.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

.emotion-7 {
  position: fixed;
  z-index: 19900410;
  min-width: 320px;
  top: 50%;
  left: 1.25em;
  -webkit-transform: translate3d(0px,-50%,0px);
  -ms-transform: translate3d(0px,-50%,0px);
  transform: translate3d(0px,-50%,0px);
}

@media screen and (max-width:320px) {
  .emotion-7 {
    min-width: unset;
    width: 100%;
  }
}

.emotion-7:focus {
  outline: none;
}

.emotion-7.emotion-7 {
  background-color: #ff0000;
}

.emotion-6 {
  box-shadow: 0px 3px 12px 0px rgba(0,0,0,0.15),0px 0px 6px 3px rgba(0,0,0,0.02);
}

.emotion-6.emotion-6 {
  position: fixed;
  background-color: white;
  left: 0;
  height: 100vh;
}

.emotion-6.emotion-6 {
  background-color: #ff0000;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-ModalDisclosure fp-DrawerDisclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-6 emotion-7 emotion-1 emotion-2 fp-Box fp-Drawer fp-Modal"
    data-dialog="true"
    id="test"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="emotion-3 emotion-1 emotion-2 fp-Box fp-ModalBackdrop"
      data-dialog-ref="test"
    >
      <div />
    </div>
    This is a side overlay
  </div>
</div>
`;
